test/exploratory/RPC.bigNumber.test.ts

```diff
@@ -0,0 +1,64 @@
+import {api, getApi, initApi} from "../../utils/api";
+import { getBalanceOfPool} from '../../utils/tx'
+import {waitNewBlock} from '../../utils/eventListeners'
+import BN from 'bn.js'
+import { Keyring } from '@polkadot/api'
+import {User} from "../../utils/User";
+import { Assets } from "../../utils/Assets";
+import { getEnvironmentRequiredVars } from "../../utils/utils";
+import { signSendAndWaitToFinishTx } from "../../utils/txHandler";
+
+
+jest.spyOn(console, 'log').mockImplementation(jest.fn());
+jest.setTimeout(15000000000);
+process.env.NODE_ENV = 'test';
+const {sudo:sudoUserName} = getEnvironmentRequiredVars();
+
+//The idea of this is to use it as a playground, so whenever its needed to test any specifics,
+// we can setup the specific scenario and test.
+// always skip the test suite to avoid long test executions.
+// node ./node_modules/jest/bin/jest.js test/exploratory/playground.ts --testRegex='.*' 
+describe('Playground', () => {
+	
+	var testUser1 : User;
+	var sudo : User;
+	var api: any;
+	var keyring : Keyring;
+
+	//creating pool
+	
+	beforeAll( async () => {
+		try {
+			getApi();
+		  } catch(e) {
+			await initApi();
+		}
+	});
+
+	beforeEach(async () => {
+		await waitNewBlock();
+		keyring = new Keyring({ type: 'sr25519' });
+	
+		// setup users
+		testUser1 = new User(keyring);
+	
+		sudo = new User(keyring, sudoUserName);
+		
+		// add users to pair.
+		keyring.addPair(testUser1.keyRingPair);
+		keyring.addPair(sudo.keyRingPair);
+		api = getApi();
+	});
+	const MAX_BALANCE = new BN('340282366920938463463374607431768211455'); //max balance
+	test('Request big numbers', async () => {
+
+		let result = await ( api.rpc as any).xyk.calculate_buy_price(MAX_BALANCE, MAX_BALANCE, MAX_BALANCE);
```

Comment:
hi @iStrike7 
When using MAX_BALANCE it in the RPC it fails badly,
```
    2021-07-09 08:17:47        RPC-CORE: calculate_buy_price(input_reserve: Balance, output_reserve: Balance, sell_amount: Balance): RpcResult<Balance>:: -32602: Invalid params: invalid type: string "0xfffffffffffffffffffffffffffffffe", expected u128.

      at apply (node_modules/@polkadot/util/logger.js:63:22)
      at Object.error (node_modules/@polkadot/util/logger.js:98:27)
      at logErrorMessage (node_modules/@polkadot/rpc-core/index.js:63:5)
      at node_modules/@polkadot/rpc-core/index.js:270:11
```

Reply:
@iStrike7 , any news from the RPCs calls? were you able to investigate it?
Thanks

Reply:
Mat fixed this :)